### DISCUSSION TITLE
How to enable color when using the function pet-select ZSH function?

### DISCUSSION THREAD

**Discussion by progS1m** [knqyf263/pet#255] [Q&A]
I've tried to add --color option to the pet-select ZSH function but it does change nothing and the color do not appear.
Tried two variants (.zshr) :
```
...
function pet-select() {
--
130 | BUFFER=$(pet search --color --query "$LBUFFER")
...
```

or

```
...
function pet-select() {
--
130 | BUFFER=$(pet search --query "$LBUFFER" --color)
...
```

Using the pet search --color brings the colorized version.

**Comment by lrworth** [comment]
I see the same thing in bash. Running `pet search --color` displays coloured output, but `$(pet search --color)` does not.

**Comment by RamiAwar** [comment]
Hmm not sure how to test this one. If I run pet without any configuration (fresh build, nothing in my environment) I get colors in both bash and zsh, but not the colors I would get if I ran `pet search --color`.

This is indeed strange

<img width="221" alt="image" src="https://github.com/knqyf263/pet/assets/8523191/f834369e-75ed-48b0-851e-9835b1c4f0ea">

<img width="154" alt="image" src="https://github.com/knqyf263/pet/assets/8523191/f5bd356b-93cc-4c04-9ae3-4de4458abe5d">

Would appreciate help looking into this 😁

**Reply by RamiAwar** [reply]
It seems as if the ZSH shell 'theme' is not there when executing the command in $(...)

**Reply by v-snap** [reply]
Hi RamiAwar , any solution to this , its very hard to read or navigate snippets when running pet-select()

**Reply by RamiAwar** [reply]
I think this depends a lot on your terminal and is very hard to debug user to user. For me for example it just works on all my machines on bash & zsh. Haven't tested on windows though.

What shell & OS are you using? @viprob-ai

**Reply by v-snap** [reply]
For me it's not working in bash shell

Shell: bash
OS : Ubuntu 22.04

I have put the `pet-select` in bashrc

**Reply by RamiAwar** [reply]
Can you post the contents of your bash profile maybe?

I'll see if copying them then helps me replicate

**Comment by v-snap** [comment]
Hi @RamiAwar  , Please find below config.toml 
[config.txt](https://github.com/user-attachments/files/17607326/config.txt)

here my bashrc file with pet-select function added at bottom
[bashrc.txt](https://github.com/user-attachments/files/17607342/bashrc.txt)

```
function pet-select() {
  BUFFER=$(pet search --color --query "$READLINE_LINE")
  READLINE_LINE=$BUFFER
  READLINE_POINT=${#BUFFER}
}
```

**pet version : 0.9.1**

pet search --color            |  pet-select
:-------------------------:|:-------------------------:
![Screenshot from 2024-11-02 20-55-18](https://github.com/user-attachments/assets/51199797-f6c7-4423-824a-c79719370673) | ![Screenshot from 2024-11-02 20-55-47](https://github.com/user-attachments/assets/2b7c45ea-7ce9-4d92-827a-c12df09283bd)

there is **No** color when invoked with pet-select
